libs/langchain-openai/src/chat_models.ts

```diff
@@ -200,7 +202,127 @@ export function _convertMessagesToOpenAIParams(
   });
 }
 
-type ChatOpenAIToolType = BindToolsInput | OpenAIClient.ChatCompletionTool;
+export function _convertMessagesToOpenAIResponsesParams(
+  messages: BaseMessage[],
+  model?: string
+): OpenAIResponsesInputItem[] {
+  return messages.flatMap(
+    (message): OpenAIResponsesInputItem | OpenAIResponsesInputItem[] => {
+      let role = messageToOpenAIRole(message);
+      if (role === "system" && isReasoningModel(model)) role = "developer";
+      if (role === "tool") {
+        const toolMessage = message as ToolMessage;
+        return {
+          type: "function_call_output",
+          call_id: toolMessage.tool_call_id,
+          id: toolMessage.id,
+          output: JSON.stringify(toolMessage.content),
+          status: "completed",
+        };
+      }
+
+      if (message.additional_kwargs.function_call != null) {
+        throw new Error("Function call is not supported in Responses API");
+      }
+      if (role === "function") {
+        throw new Error("Function messages are not supported in Responses API");
+      }
+
+      if (isAIMessage(message) && !!message.tool_calls?.length) {
+        return message.tool_calls.map(
+          (toolCall): OpenAIResponsesInputItem => ({
+            type: "function_call",
+            name: toolCall.name,
+            arguments: JSON.stringify(toolCall.args),
+            call_id: toolCall.id!,
+            id: toolCall.id!,
```

Comment:
these are different i believe. 'id' is the object id, 'call_id' is what you'll use to refer to this tool call when returning the output

![Screenshot 2025-03-11 at 5 58 04 PM](https://github.com/user-attachments/assets/f685c9a3-04a7-4398-96b9-30f778d33d51)

Reply:
'call_id' maps to what we store as toolCall.id

Reply:
for now was thinking of throwing the new 'id' for all tool calls in AIMessage.additionalKwargs['openai_tool_call_ids'] or something like that

Reply:
smth like
```python
            if tool_calls := lc_msg.tool_calls:
                if not lc_msg.additional_kwargs.get("__openai_function_call_ids__"):
                    raise ValueError(...)
                function_call_ids = lc_msg.additional_kwargs["__openai_function_call_ids__"]
                for tool_call in tool_calls:
                    function_call = {
                        "type": "function_call",
                        "name": tool_call["name"],
                        "arguments": tool_call["arguments"],
                        "call_id": tool_call["id"],
                        "id": function_call_ids[tool_call["id"]],
                    }
                    updated_payload["input"].append(function_call)

```